### ISSUE TITLE
More automatic translation of is.pure.levelX -> label?

### ISSUE THREAD

**Issue by bbimber** [carmonalab/scGate#2]
Hello,

We're very interested in scGate. As far as I understand it, is you give it a hierarchical gate model, you end up with a bunch of columns corresponding to the cell classes (i.e. Bcell_UCell, Tcell_UCell), and then a series of binary is.pure.levelX columns. Is there a helper that more automatically translates from is.pure.levelX into the cell type label that this cell received from levelX?

Thanks

**Comment by mass-a** [comment]
Hello and thanks for your interest in the tool!

scGate is a binary classifier (pure vs. non-pure for a target cell type) rather than a multi-class classifier. So at each level of a hierarchical model, a binary decision is made on whether a given cell is "pure" for that particular level. This information is stored in the is.pure.levelX column, for each level of the gating model, but this is a binary assignment to pure/non-pure, rather than a label to a specific cell type.

I agree it would be helpful to understand at least qualitatively why certain groups of cells were labeled as impure. What we normally do is to systematically plot (e.g. over a UMAP representation of the data) the UCell scores stored in the output object, to highlight signatures that characterize groups of cells. But suggestions to automatize this are welcome!

Cheers,
-m

**Comment by bbimber** [comment]
Thanks for the fast reply. I understand the binary classifier aspect, and the single 'is.pure' column makes a ton of sense for something like your single population, NK cell example. 'is.pure' always refers to the label 'NK'
```
my_scGate_model <- gating_model(name = "NK", signature = c("KLRD1","CD3D-"))  
```
in contrast, when you have a hierarchical gate, being pure or impure at a given level maps to a label. you do this in figure D here: https://github.com/carmonalab/scGate. Also, your gate model should hold that translation, like you show with the gate visualization here:

https://carmonalab.github.io/scGate.demo/#visualizing-gating-models

Using your  models.DB$human$generic$Plasma_cell example, then wouldnt the translations be:

is.pure.level1 == 'Pure' -> 'Immune Lymphoid PanBcell Plasma_cell'
is.pure.level1 == 'Impure' -> 'Epithelial Stromal'

Basically 'positive' on the tree equates to 'pure' in the field, correct? It would be helpful if code could take a gate model and a seurat object and automatically do this re-labeling.

Stepping back: my lab is thinking about the problem of how to take a heterogeneous sample, like PBMC, and run a gate or set of gates to categorize all the major cell types. Your reply above and reading your docs makes me think that your goal is more about executing scGate to identify a single population (i.e. find plasma cells, but dont bother categorizing the rest). Is that a fair statement?

**Comment by mass-a** [comment]
Ok I get it now. You are suggesting to translate the pure/impure assignments at each level to the positive/negative cell types used for that assignment. I agree it would be useful to show this information more directly, we will think about how to implement it.

> Stepping back: my lab is thinking about the problem of how to take a heterogeneous sample, like PBMC, and run a gate or set of gates to categorize all the major cell types. Your reply above and reading your docs makes me think that your goal is more about executing scGate to identify a single population (i.e. find plasma cells, but dont bother categorizing the rest). Is that a fair statement?

Yes that is correct. There exist methods aimed at classifying all cell types in a sample, but it's a difficult task and performance is often not very good. scGate takes a more limited scope of identifying a single cell population with high precision, to enable downstream analyses for that particular cell type. Of course nothing prevents you from running several gating models for each of a series of cell types in a PBMC sample, and in this way isolate individual cell types. The challenge in this case is to have an accurate gating model for each cell type of interest.

**Comment by sjcarmona** [comment]
Hi @bbimber, you can try a new scGate's feature currently in **dev** (`install_github("https://github.com/carmonalab/STACAS",ref="dev")`) to efficiently evaluate multiple models and generate a multi-class output with the model names. You can just pass a named list of models as input for _model_ param, and you will get your result in a new meta.data column called _scGate_multi_. If a cell was _pure_ for a single model,  _scGate_multi_ will contain the name of that model (eg. NK); if a cell was _pure_ for more than one model,  _scGate_multi_ will be _multi_, or _NA_ otherwise. Thanks for your feedback

**Comment by bbimber** [comment]
Thanks, we'll try to have a look. Did you mean to link to the STACAS repo and not scGate itself?

**Comment by sjcarmona** [comment]
No, sorry, scGate dev
